src/main/java/org/icgc/argo/program_service/services/EgoAuthorizationService.java

```diff
@@ -0,0 +1,62 @@
+package org.icgc.argo.program_service.services;
+
+import io.grpc.Status;
+import lombok.val;
+import org.icgc.argo.program_service.grpc.interceptor.EgoAuthInterceptor;
+import org.icgc.argo.program_service.services.ego.model.entity.EgoToken;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+@Profile("auth")
+@Service
+public class EgoAuthorizationService implements AuthorizationService {
+  public EgoToken getEgoToken() {
+    return EgoAuthInterceptor.EGO_TOKEN.get();
+  }
+
+  private EgoToken fromToken() {
+    val token = getEgoToken();
+    if (token == null) {
+      throw Status.fromCode(Status.Code.UNAUTHENTICATED).asRuntimeException();
+    }
+    return token;
+  }
+
+  public boolean hasPermission(String permission) {
+    return getPermissions().contains(permission);
+  }
+
+  @Override
+  public boolean isDCCAdmin(){
+    val type = fromToken().getType();
+
+    if (type == null) {
+      return false;
+    }
+    return type.equalsIgnoreCase("ADMIN");
```

Comment:
I thought we decided to use a PROGRAM_SERVICE policy to identify someone with DCC permissions. Using Ego user types outside of Ego scope conflates the meaning of that user type between other services and someone that specifically has full ego access... Its best to keep these as separate concepts...